packages/protocol/contracts/layer1/based/TaikoInbox.sol

```diff
@@ -536,8 +536,9 @@ abstract contract TaikoInbox is EssentialContract, ITaikoInbox, ITaiko, IFork {
 
         SyncBlock memory synced;
 
-        uint256 stopBatchId = (_config.maxBatchesToVerify * _length + _stats2.lastVerifiedBatchId)
-            .min(_stats2.numBatches);
```

Comment:
could you please add a unit test to verify the bug?

Reply:
Ok, test cases will be added later and it's a good opportunity to get familiar with the testing framework. :)